src/integer_mod_q/poly_over_zq/from.rs

```diff
@@ -270,3 +315,24 @@ mod test_from_str {
         );
     }
 }
+
+#[cfg(test)]
+mod test_from_modulus_polynomial_ring_zq {
+    use crate::integer_mod_q::{ModulusPolynomialRingZq, PolyOverZq};
+    use std::str::FromStr;
+
+    /// ensure that the conversion works with positive large entries
+    #[test]
+    fn large_positive() {
+        let modulus_ring =
+            ModulusPolynomialRingZq::from_str(&format!("4  -1 0 0 1 mod {}", u64::MAX - 58))
+                .unwrap();
+
+        let modulus = PolyOverZq::from(&modulus_ring);
+
+        let cmp_poly =
+            PolyOverZq::from_str(&format!("4  {} 0 0 1 mod {}", u64::MAX - 59, u64::MAX - 58))
```

Comment:
Why do you use `u64::MAX-58` instead of `u64::MAX` for the modulus?

Reply:
its prime, I made it clearer in the code